File: Solutions/0921/0921.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/10/4 11:01
# File: 0921.py
# Desc: 


class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        l = []
        for i in s:
            if i == ')':
                if l and l[-1] == '(': 
                    l.pop()
                    continue
            l.append(i)
        return len(l)

File: Solutions/0921/0921.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 0921.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-10-4 11:01:36
***************************************************************************************************/

int minAddToMakeValid(char * s){
    char * l = (char *)malloc(sizeof(char) * strlen(s));
    int idx = 0;
    for (int i=0; i<strlen(s); i++) {
        if (s[i] == ')') {
            if (idx>0 && l[idx-1] == '(') {
                idx--;
                l[idx] = '-';
                continue;
            }
        }
        l[idx++] = s[i];
    }
    return idx;
}
